optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -0,0 +1,1192 @@
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    token=token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                file_names[name] = model_cache_path
+
+            model_save_dir = Path(model_cache_path).parent
+            if not compile_only:
+                language_model = model_cls.load_model(file_names["language_model"], quantization_config)
+                text_embeddings = model_cls.load_model(file_names["text_embeddings"], quantization_config)
+                vision_embeddings = model_cls.load_model(file_names["vision_emnbeddings"], quantization_config)
+                for part in model_cls.additional_parts:
+                    kwargs[part] = model_cls.load_model(file_names[part], quantization_config)
+            else:
+                language_model = model_cls._compile_model(
+                    file_names["language_model"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                text_embeddings = model_cls._compile_model(
+                    file_names["text_embeddings"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                vision_embeddings = model_cls._compile_model(
+                    file_names["vision_embeddings"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                for part in model_cls.additional_parts:
+                    kwargs[part] = model_cls._compile_model(
+                        file_names[part],
+                        kwargs.get("device", "CPU"),
+                        kwargs.get("ov_config"),
+                        model_save_dir,
+                    )
+        try:
+            generation_config = GenerationConfig.from_pretrained(
+                model_id,
+                token=token,
+                revision=revision,
+                cache_dir=cache_dir,
+                force_download=force_download,
+                local_files_only=local_files_only,
+            )
+            kwargs["generation_config"] = generation_config
+        except Exception:
+            pass
+
+        return model_cls(
+            language_model=language_model,
+            text_embeddings=text_embeddings,
+            vision_embeddings=vision_embeddings,
+            config=config,
+            model_save_dir=model_save_dir,
+            quantization_config=quantization_config,
+            **kwargs,
+        )
+
+    @classmethod
+    def _from_transformers(
```

Comment:
what is the main difference with https://github.com/huggingface/optimum-intel/blob/6a8718b76a5fc5e6796483746eded64e4007e94d/optimum/intel/openvino/modeling_decoder.py#L262 ? If only the "with-past" suffix maybe we can change it there instead wdyt ?

Reply:
do you mean I should inherit from OVModelForCausalLM to get this behaviour? I think it may lead to issues with quantization as it may requires slightly different approach for doing that (difference in input data collection, different approaches for vision embeddings and text model and so on)

Reply:
makes sense, let's keep it that way then!